### DISCUSSION TITLE
Installing the Unified Planning Library on Linux

### DISCUSSION THREAD

**Discussion by Yomaster10** [aiplan4eu/unified-planning#271] [Q&A] [answered]
How do I install the UPF on Linux (in either Ubuntu 18 or 20)?

**Comment by Framba-Luca** [comment, marked as answer]
The simplest way to install the UP library is to use pip.
`pip install unified-planning` is the command you are searching for.

If you also want to install some specific engines, you can use `pip install unified-planning[pyperplan,tamer]`. If this doesn't work for some engines, it means their integration is still under development.

To find the available engines, you can go [here](https://github.com/aiplan4eu) and search for folders named `up-{engine_name}`, like `up-tamer`, `up-enhsp` or `up-fast-downward`.
